narwhals/dataframe.py

```diff
@@ -408,16 +408,21 @@ def shape(self) -> tuple[int, int]:
         return self._compliant_frame.shape  # type: ignore[no-any-return]
 
     @overload
-    def __getitem__(self, item: Sequence[int]) -> Series: ...
+    def __getitem__(self, item: tuple[Sequence[int], str | int]) -> Series: ...  # type: ignore[overload-overlap]
+
+    @overload
+    def __getitem__(self, item: Sequence[int]) -> Self: ...
 
     @overload
     def __getitem__(self, item: str) -> Series: ...
 
     @overload
     def __getitem__(self, item: slice) -> Self: ...
 
-    def __getitem__(self, item: str | slice | Sequence[int]) -> Series | Self:
-        if isinstance(item, str):
+    def __getitem__(
+        self, item: str | slice | Sequence[int] | tuple[Sequence[int], str | int]
```

Comment:
Typing here is getting a bit crowded, should we have [something similar to `polars`](https://github.com/pola-rs/polars/blob/main/py-polars/polars/_typing.py#L256)?

Reply:
sure (not a blocker though)